script.js

```diff
@@ -29,4 +49,126 @@ function operate(operator, a, b) {
+}
+
+function doesArrayContainOperator(array) {
+    const [possibleOperator] = array;
+    return OPERATORS.includes(possibleOperator);
+}
+
+// After calculation is done, clicking on another button will clear the previous information and also the display
+const buttons = document.querySelectorAll("button");
+buttons.forEach( (button) => {
+    button.addEventListener("click", (e) => {
+        if (currentlyHeldOperator === FINISHED_CALCULATION) {
+            clearAll()
+        }
+    })
+})
+
+const clearButton = document.querySelector("#clear");
+clearButton.addEventListener("click", (e) => {
+    clearAll();
+})
+
+const digitButtons = document.querySelectorAll(".digit");
+digitButtons.forEach( (digitButton) => {
+    digitButton.addEventListener("click", (e) => {
+        const digit = e.target.getAttribute("data-value")
+        if (doesArrayContainOperator(buffer)) {
+            buffer = [digit]
+            addToInputDisplay(digit, inputsForDisplay);
+        } else if (buffer.length == 0 || NUMBERS.includes(buffer[buffer.length-1])) {
+            buffer.push(digit);
+            addToInputDisplay(digit, inputsForDisplay);
+        }
+
+    })
+})
+
+const operatorButtons = document.querySelectorAll(".operator");
+operatorButtons.forEach( (operatorButton) => {
+    operatorButton.addEventListener("click", (e) => {
+        const symbol = e.target.getAttribute("data-value")
+        if (buffer.length === 0) {
+            // If no numbers were chosen before hitting operator button, display syntax error
+            clearAll();
+            updateResultDisplay(SYNTAX_ERROR_MESSAGE);
+        } else if (doesArrayContainOperator(buffer)) {
+            // If "characters" includes an operator, update chosen operator
+            buffer = [symbol];
+            currentlyHeldOperator = symbol;
+            inputsForDisplay.splice(-1,1,` ${symbol} `)
+            updateInputDisplay(inputsForDisplay)
+        } else if (currentlyHeldOperator) {
+            // For the nth operator selection (when there is an operator in the currentlyHeldOperator)
+            reversePolishNotation.push(buffer.join(""));
+            reversePolishNotation.push(currentlyHeldOperator);
+            buffer = [symbol];
+            currentlyHeldOperator = symbol;
+            addToInputDisplay(` ${symbol} `, inputsForDisplay);
+        } else {
+            // The first operator selection (aka when currentlyHeldOperator = null)
+            reversePolishNotation.push(buffer.join(""));
+            buffer = [symbol];
+            currentlyHeldOperator = symbol;
+            addToInputDisplay(` ${symbol} `, inputsForDisplay);
+        }
+    })
+})
+
+const equalButton = document.querySelector(".equal");
+equalButton.addEventListener("click", (e) => {
+    if (doesArrayContainOperator(buffer)) {
+        updateResultDisplay(MISSING_NUMBERS_MESSAGE)
+        return
+    }
+    reversePolishNotation.push(buffer.join(""));
```

Comment:
You can combine these two push statements into one. Push can take multiple arguments to push.